pkgs/by-name/nf/nfd/package.nix

```
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  boost,
+  fetchFromGitHub,
+  libpcap,
+  ndn-cxx,
+  openssl,
+  pkg-config,
+  sphinx,
+  systemd,
+  wafHook,
+  websocketpp,
+  withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemd,
+  withWebSocket ? true,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nfd";
+  version = "24.07";
+
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = "NFD";
+    rev = "NFD-${version}";
+    hash = "sha256-HbKPO3gwQWOZf4QZE+N7tAiqsNl1GrcwE4EUGjWmf5s=";
+  };
+
+  prePatch = ''
```

```suggestion
  prePatch = lib.optionalString withWebSocket ''
```
or does that break things?

---

It's optional, so i can add that `optionalString` in the prePatch. Let me fix this